content/en/apps/guides/hosting/4.x/data-migration.md

```diff
@@ -0,0 +1,244 @@
+
+{{% alert title="Note" %}} If this step is omitted, 4.x API will fail to respond to requests until all views are indexed. Depending on the size of the database, this could take many hours, or even days. {{% /alert %}} 
+
+### 3. Save existent CouchDb configuration
+
+Some CouchDb configuration values must be ported from existent CouchDb to the 4.x installation. Store them in a safe location before shutting down 3.x CouchDb.
+
+##### a. CouchDB secret 
+Used in encrypting all CouchDb passwords and session tokens.
+##### b. CouchDb server uuid
+Used in generating replication checkpointer documents, which track where replication progress between every client and the server, and ensure that clients don't re-download or re-upload documents.
+
+Use the migration tool to obtain these values.
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration get-env
+```
+
+### 4. Locate and backup CouchDb Data folder
+a) If running in MedicOS, [CouchDb data folder]({{< relref "apps/guides/hosting/3.x/self-hosting#backup" >}}) can be found at `/srv/storage/medic-core/couchdb/data`.
+
+b) If running a custom installation of CouchDb, data would be typically stored at `/opt/couchdb/data`.
+
+c) TODO when using orchestration / AWS 
+
+### 5. Launch 4.x CouchDb installation
+
+Depending on your project scalability needs and technical possibilities, you must decide whether you will deploy CouchDb in a single node or in a cluster with multiple nodes.
+Please consult this guide about clustering and horizontal scalability to make an informed decision. <insert link>
+
+{{% alert title="Note" %}} You can start with single node and then change to a cluster. This involves running the migration tool again to distribute shards from the existent node to the new nodes. {{% /alert %}}
+
+Depending on your choice, follow the instructions that match your deployment below: 
+
+#### Single node
+
+a) Download 4.x single-node CouchDb docker-compose file:
+```shell
+mkdir -p ~/couchdb-single/ 
+cd ~/couchdb-single/ 
+curl -s -o ./docker-compose.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.0/docker-compose/cht-couchdb.yml
+```
+a) Make a copy of the 3.x CouchDb data folder from **step 4**.
+
+b) Set the correct environment variables:
+```shell
+cat > ${HOME}/couchdb-single/.env << EOF
+COUCHDB_USER=<admin>
+COUCHDB_PASSWORD=<password>
+COUCHDB_SECRET=<COUCHDB_SECRET from step 3>
+COUCHDB_UUID=<COUCHDB_UUID from step 3>
+COUCHDB_DATA=<absolute path to folder created in step 5.a>
+EOF
+```
+c) Start 4.x CouchDb and wait until it is up.
+```shell
+cd ~/couchdb-single/ 
+docker-compose up -d
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration check-couchdb-up
+```
+
+d) Change metadata to match the new CouchDb node
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration move-node
+```
+e) Verify that the migration was successful
+```shell
+docker-compose run couch-migration verify
+```
```

Comment:
same comment as above - we should  qualify how they know `move-node` and `verify` calls correctly executed.

Reply:
Suggestions appreciated :) 
The command just completes without error at the moment. Should we have a success message at the end?

Reply:
OMG - these two and `check-couchdb-up` all having a "$command complete" (or something) at the end would be dreamy!  If it's not too late, that'd be my vote.  Then we could have:

> Verify that the migration was successful burn running this and waiting for it to exit with `check-couchdb-up complete`

Otherwise, like above we did:

> Verify that the migration was successful burn running this `docker-compose` command and waiting for it to exit with no errors:`

Reply:
I believe success messages were added in #medic/couchdb-migration/pull/7, so we can now update this section with the corresponding messages

Reply:
i see these were added - thanks!